src/faostat_data_primap/download.py

```diff
@@ -1,20 +1,162 @@
+    all_releases = [
+        release_name
+        for release_name in os.listdir(current_release_path.parent)
+        if (domain_path / release_name).is_dir()
+    ]
+
+    # make sure all directories follow the naming convention
+    try:
+        all_releases_datetime = [
+            datetime.strptime(release, "%Y-%m-%d") for release in all_releases
+        ]
+    except ValueError as e:
+        msg = (
+            "All release folders must be in YYYY-MM-DD format, "
+            f"got {sorted(all_releases)}"
+        )
+        raise ValueError(msg) from e
+
+    all_releases_datetime = sorted(all_releases_datetime)
+    current_release_datetime = datetime.strptime(current_release_path.name, "%Y-%m-%d")
+    index = all_releases_datetime.index(current_release_datetime)
+
+    # if the current release is the latest or the only one
+    if index == 0:
+        return None
+
+    return domain_path / all_releases_datetime[index - 1].strftime("%Y-%m-%d")
+
+
+def calculate_checksum(file_path: pathlib.Path) -> str:
+    """
+    Calculate the SHA-256 checksum of a file.
+
+    Parameters
+    ----------
+    file_path : pathlib.Path
+        The path to the file for which the checksum is calculated.
+
+    Returns
+    -------
+    str
+        The SHA-256 checksum of the file as a hexadecimal string.
+    """
+    sha256 = hashlib.sha256()
+    with open(file_path, "rb") as f:
+        for chunk in iter(lambda: f.read(4096), b""):
+            sha256.update(chunk)
+    return sha256.hexdigest()
+
+
+def download_methodology(url_download: str, save_path: pathlib.Path) -> None:
+    """
+    Download methodology file.
+
+    Download the methodology PDF-file from a specified URL and save to a
+    target directory. If the file already exists in `save_path`,
+    the download is skipped. If a previous release directory exists,
+    the function attempts to locate the file there and compares checksums
+    to avoid downloading an identical file. If it exists in the previous release,
+    but it's not identical it is downloaded. If the file exists in the previous
+    release directory and is identical, a symlink will be created instead of downloading
+    to avoid duplicate downloads. If the file does not exist in a previous release,
+    it will be downloaded.
+
+    Parameters
+    ----------
+    url_download : str
+        The URL from which to download the file.
+    save_path : pathlib.Path
+        The path to the directory where the file should be saved.
+    """
+    filename = url_download.split("/")[-1]
+    download_path = save_path / filename
+
+    if download_path.exists():
+        print(f"Skipping download of {download_path} because it already exists.")
+        return
+
+    previous_release = find_previous_release_path(save_path)
+    # Attempt to find a file to compare in the previous release
+    if previous_release:
+        file_to_compare = previous_release / filename
+        if file_to_compare.exists():
```

Comment:
If a file can not be downloaded in the 30s period the script will fail, correct? I thin this is fine for now, but might need changes if the FAO servers are not always fast with responses (using multiple tries).